### ISSUE TITLE
Feature: Better Serializer exception handling

### ISSUE THREAD

**Issue by oxplay2** [tlf30/monkey-netty#14]
When i started using Netty directly(trying Object DEMO), i had issue that one of objects within objects within objects was not serialiable.

and i didnt receive ANY information about issue. There were just no info.
So i reported it in Netty, and now they made PR(already merged) that will propagate all missing exceptions about Serialization message.

see:

https://github.com/netty/netty/pull/10807/commits/08b496140139544aa945ddd74db9080a7445fd47

I think we should add same change into our handler, so people that use Monkey-Netty will know they "for example forgot add Serializable somewhere".

What do you think? 

We could add Setting for this like "debugMode" and add additional pipeline listener like in PR when its enabled.

**Comment by tlf30** [comment]
Yeah, this should be straightforward. I will add a check to the NetworkMessageOutputStream.

**Comment by tlf30** [comment]
This will be waiting on the netty.io PR. I have implemented the changes on this side, but until the exception is propagated up, I have no way to throw an exception myself in the code and have it be visible to the user.

**Comment by tlf30** [comment]
I have pushed a patch to master for this, I will leave this open until we can get netty updated with the fix on their side, and tested with it.

**Comment by tlf30** [comment]
OK, I have this working:
```
io.netty.handler.codec.EncoderException: io.tlf.monkeynetty.NetworkMessageException: Non-Serializable object io.tlf.monkeynetty.test.messages.NonSerial found in message io.tlf.monkeynetty.test.messages.TestUDPMessage
	at io.netty.handler.codec.MessageToByteEncoder.write(MessageToByteEncoder.java:125)
	at io.netty.channel.AbstractChannelHandlerContext.invokeWrite0(AbstractChannelHandlerContext.java:717)
	at io.netty.channel.AbstractChannelHandlerContext.invokeWriteAndFlush(AbstractChannelHandlerContext.java:764)
	at io.netty.channel.AbstractChannelHandlerContext$WriteTask.run(AbstractChannelHandlerContext.java:1071)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:164)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:472)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:500)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: io.tlf.monkeynetty.NetworkMessageException: Non-Serializable object io.tlf.monkeynetty.test.messages.NonSerial found in message io.tlf.monkeynetty.test.messages.TestUDPMessage
	at io.tlf.monkeynetty.NetworkMessageEncoder.encode(NetworkMessageEncoder.java:61)
	at io.tlf.monkeynetty.NetworkMessageEncoder.encode(NetworkMessageEncoder.java:44)
	at io.netty.handler.codec.MessageToByteEncoder.write(MessageToByteEncoder.java:107)
	... 10 more
```

**Comment by tlf30** [comment]
I think it would be good to have this enabled at all times

**Comment by oxplay2** [comment]
if it will not slow-down process, then sure :) can stay as pernament addition.

**Comment by tlf30** [comment]
Resolved in #16
